libsupport/include/katana/URI.h

```diff
@@ -63,6 +63,9 @@ class KATANA_EXPORT URI {
 
   URI operator+(char rhs) const;
   URI operator+(const std::string& rhs) const;
+  bool operator<(URI const& o) const {
+    return path_ < o.path_ || (path_ == o.path_ && scheme_ < o.scheme_);
+  }
```

Comment:
I would expect scheme to come first so this is closer to lexical order.

Reply:
Yeah, I actually thought about this a little.  I figured file://a < http://b is more intuitive because the scheme is almost out of band information and not lexicographically meaningful.

But your suggestion is reasonable.  I was also worried if file:// schemes are sometimes represented as empty.  That doesn't look to be a problem.

Reply:
`file://` isn't a valid URL, right? File URLs must always be absolute and are written with an empty host part, so they always start with `file:///` (*three* slashes not two).

Reply:
Random guy on the internet agrees with me: https://stackoverflow.com/questions/7857416/file-uri-scheme-and-relative-files

Reply:
I meant to write this.  Sorry to confuse.

Yeah, I actually thought about this a little. I figured file:///a < http:///b is more intuitive because the scheme is almost out of band information and not lexicographically meaningful.

But your suggestion is reasonable. I was also worried if file:/// schemes are sometimes represented as empty. That doesn't look to be a problem.

Reply:
The danger of sentinels that are part of the type. :-) With correct use of `std::optional` that shouldn't be an issue anyway.